### ISSUE TITLE
http: Proxy environment variables are ignored

### ISSUE THREAD

**Issue by spuranam** [inspec/inspec#765]
### Description

Inspec compliance subcommands does not honor http_proxy, no_proxy environment variables.
### InSpec and Platform Version

0.22.1
### Replication Case

Tell us what steps to take to replicate your problem.  See [How to create a Minimal, Complete, and Verifiable example](https://stackoverflow.com/help/mcve)
for information on how to create a good replication case.
### Possible Solutions

If you have already ideas how to solve the issue, add them here.
### Stacktrace

Please include the stacktrace output or link to a gist of it, if there is one.

**Comment by chris-rock** [comment]
That is a great idea @spuranam I mark this as a feature request.

**Comment by geoffles** [comment]
+1 

Would also be convenient to specify a no proxy at all on the http resource.

**Comment by jblaine** [comment]
Likewise, the `inspec supermarket...` command does not pay any attention to typical proxy environment variables.
```
[@gazoo:~/rcf-chef] master+ ± inspec supermarket exec cis-linux-benchmark
/opt/chefdk/embedded/lib/ruby/2.4.0/net/http.rb:906:in `rescue in block in connect': Failed to open TCP connection to supermarket.chef.io:443 (Connection refused - connect(2) for "supermarket.chef.io" port 443) (Errno::ECONNREFUSED)
```

**Comment by chrisingenhaag** [comment]
+1

**Comment by justjoehere** [comment]
Has anyone figured out a way to make this work after 2 years?   Or should I start looking for another tool to do infrastructure and compliance validations?

**Comment by aaronlippold** [comment]
I believe this has been worked with our other Proxy Pass and Bastion Host support in inspec and train. However, if not, please provide more feedback so we can put it on the 3.0.x backlog as a bug.

**Comment by tbugfinder** [comment]
on my end, common PROXY settings are not used by inspec:
```
$ bundle exec inspec supermarket profiles
bundler: failed to load command: inspec (/REPO/repo/vendor/bundle/ruby/2.3.0/bin/inspec)
Net::OpenTimeout: execution expired

$ bundle exec inspec version
2.3.24

```

**Comment by nellshamrell** [comment]
Looking into this

**Comment by nellshamrell** [comment]
I'm having some trouble replicating this:

I set up a workstation server and a proxy server (the workstation only allowed outbound traffic to the proxy server) using this Terraform config https://github.com/nellshamrell/tf_workstation_to_proxy

The Terraform config also sets http_proxy to the proxy_server and appropriate port on the workstation.

I then installed inspec on the workstation and ran 

```
(workstation) $ inspec supermarket exec dev-sec/linux-baseline
```

And tracked the proxy traffic (the proxy was set up with squid).

The request showed as coming into the proxy, was received, the proxy reached out to supermarket, then passed the traffic back to the workstation. The linux-baseline profile ran just fine on the workstation.

I'm wondering if this may have been fixed at some point in the past - it's the end of the work day here, but I'll check that out tomorrow.

In the meantime, can anyone else still replicate this issue?
